hostinfo/mysql.lua

```diff
@@ -0,0 +1,182 @@
+local HostInfo = require('./base').HostInfo
+local Transform = require('stream').Transform
+local misc = require('./misc')
+local run = misc.run
+local safeMerge = misc.safeMerge
+local async = require('async')
+--------------------------------------------------------------------------------------------------------------------
+local Reader = Transform:extend()
+
+function Reader:initialize()
+  Transform.initialize(self, {objectMode = true})
+  self._pushed = false
+end
+
+local AdminStatusReader = Reader:extend()
+function AdminStatusReader:_transform(line, cb)
+  line:gsub('%s*(%D+)%:%s(%d+)', function(key, value)
+    self:push({[key] = value})
+  end)
+  cb()
+end
+
+local KeyValueReader = Reader:extend()
+function KeyValueReader:_transform(line, cb)
+  local key, value = line:match('(%S+)[%s\t]?(%S*)')
+  self:push({[key] = value or ''})
+  cb()
+end
+
+local ReplicantUserReader = Reader:extend()
+function ReplicantUserReader:_transform(line, cb)
+  local key = line:match('(%S+)[%s\t]?(%S*)')
+  if not self._pushed then
+    if key:match('^repl') then
+      self:push('true')
+      self._pushed = true
+    end
+  end
+  cb()
+end
+
+local ProcsReader = Reader:extend()
+function ProcsReader:_transform(line, cb)
+  if not line:find('^%*') then
+    local key, value = line:match('(%S+)%:%s(.*)')
+    if key and value then
+      self:push({[key] = value})
+    end
+  end
+  cb()
+end
+
+local VersionReader = Reader:extend()
+function VersionReader:_transform(line, cb)
+  local dataTable = {}
+  line:gsub('%S+', function(c) table.insert(dataTable, c) end)
+  self:push({
+    name = line:find('MariaDB') and 'MariaDB' or 'MySql',
+    version = dataTable[3],
+    distribution = dataTable[5],
+    targetOS = dataTable[7],
+    targetArch = dataTable[8]:match('%((%S+)%)')
+  })
+  cb()
+end
+
+-----------------------------------------------------------------------------------------------------------------------
+--[[ MySQL ]]--
+local Info = HostInfo:extend()
+function Info:initialize()
+  HostInfo.initialize(self)
+end
+
+function Info:_run(callback)
+  local outTable, errTable = {}, {}
+  local mySqlServerBin = '/usr/sbin/mysqld'
```

Comment:
can we use just mysqld, mysql, and mysqladmin... if they are not installed in these locations